MetaMask/metamask-extension
# Adopt security provider request from core

## Explanation

This PR aims to adopt the security provider request from message-manager pkg (core) into `SignController`.

Additionally, migrating to ts and adding unit tests to security provider helper.

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/414

## Manual Testing Steps

- Go to [dapp test page](https://metamask.github.io/test-dapp/)
- Try to connect
- Sign a Personal message (or any other)
- Should not display any warning cause `metamask.github.io` is not flagged as malicious.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
